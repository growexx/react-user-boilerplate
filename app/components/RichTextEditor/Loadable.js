/**
 *
 * Asynchronously loads the component for RichTextEditor
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
