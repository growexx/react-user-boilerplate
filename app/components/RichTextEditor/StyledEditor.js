import styled from 'styled-components';

export const StyledEditor = styled.div`
  border: 1px solid @border-color-base;
  min-height: @height-lg + @height-sm;
  cursor: text;
`;
