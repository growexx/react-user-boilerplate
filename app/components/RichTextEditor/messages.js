/*
 * RichTextEditor Messages
 *
 * This contains all the text for the RichTextEditor component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.RichTextEditor';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the RichTextEditor component!',
  },
});
