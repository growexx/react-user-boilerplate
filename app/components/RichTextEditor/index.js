/**
 *
 * RichTextEditor
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Editor } from 'react-draft-wysiwyg';
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';

import { StyledEditor } from './StyledEditor';

class RichTextEditor extends React.Component {
  constructor(props) {
    super(props);
    this.setEditor = editor => {
      this.editor = editor;
    };
  }

  render() {
    return (
      <StyledEditor>
        <Editor
          ref={this.setEditor}
          editorState={this.props.value}
          onEditorStateChange={this.props.onChange}
        />
      </StyledEditor>
    );
  }
}

RichTextEditor.propTypes = {
  value: PropTypes.object,
  onChange: PropTypes.func,
};

export default RichTextEditor;
