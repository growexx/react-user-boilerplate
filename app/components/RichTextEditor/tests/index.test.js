/**
 *
 * Tests for RichTextEditor
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { EditorState } from 'draft-js';
import { ConnectedRouter } from 'connected-react-router';
import { browserHistory } from 'react-router-dom';
import history from 'utils/history';
import RichTextEditor from '../index';
import configureStore from '../../../configureStore';
jest.mock('draft-js/lib/generateRandomKey', () => () => '123');
let store;
const props = {
  value: EditorState.createEmpty(),
  onChange: jest.fn(),
};

const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <RichTextEditor {...props} />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<RichTextEditor />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  it('Should render and match the snapshot', () => {
    jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789);
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
});
