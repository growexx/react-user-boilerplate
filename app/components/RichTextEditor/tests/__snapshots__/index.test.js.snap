// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichTextEditor /> Should render and match the snapshot 1`] = `
<div
  class="StyledEditor-l6ccd5-0 gzMMsh"
>
  <div
    class="DraftEditor-root"
  >
    <div
      class="DraftEditor-editorContainer"
    >
      <div
        class="notranslate public-DraftEditor-content"
        contenteditable="true"
        role="textbox"
        spellcheck="false"
        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
      >
        <div
          data-contents="true"
        >
          <div
            class=""
            data-block="true"
            data-editor="aggaf"
            data-offset-key="3oi9-0-0"
          >
            <div
              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
              data-offset-key="3oi9-0-0"
            >
              <span
                data-offset-key="3oi9-0-0"
              >
                <br
                  data-text="true"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
